### PR TITLE
U-Boot boot commands Support

### PR DESCRIPTION
**Pull request by jremmet**
uboot doesn't have pre defined boot sources. The "boot" command takes no parameters.
Add self.boot_commands to hold different boot_command by name.
They can be used with LinuxBootProtocol boot(name)

**Checklist**
- [X] Tests for the feature 
- [X] PR has been tested

### CODE REVIEW

File: labgrid/driver/ubootdriver.py [hunk 1]
```diff
@@ -206,13 +206,13 @@ def await_boot(self):
         self.console.expect(self.bootstring, timeout=self.boot_timeout)
 
     @Driver.check_active
-    @step(args=['name'])
-    def boot(self, name):
-        """Boot the default or a specific boot entry
+    @step(args=['cmd'])
+    def boot(self, cmd: str = ""):
+        """Boot the default if cmd is empty or a specific boot cmd otherwise.
```
Comment (jluebbe) [inline labgrid/driver/ubootdriver.py:211]: The LinuxBootProtocol defines this parameter as `cmd`: https://github.com/labgrid-project/labgrid/blob/master/labgrid/protocol/linuxbootprotocol.py#L6

We could have a dict in the UBootDriver configuration which defines commands for each boot "entry". What do you think? @jremmet @Emantor

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: labgrid/driver/ubootdriver.py (hunk 1 above)
Comment (jremmet) [reply labgrid/driver/ubootdriver.py:211]: I like the idea. I would be configurable, and we can check if exists. Guess `self.boot_command`should continue to work and the default for `name = ""` ? @jluebbe

Comment (jluebbe) [reply labgrid/driver/ubootdriver.py:211]: Yes. Perhaps something like:
```
boot_commands:
  A: "run boot_slot_a"
  B: "run boot_slot_b"
```

Comment (jluebbe) [reply labgrid/driver/ubootdriver.py:211]: This way, you could also reuse test cases and Strategies easier with different boards.

**Review by Emantor — APPROVED** [review]

**Review by jluebbe — APPROVED** [review]

**Pull request merged**